src/core/agent/agent.ts

```diff
@@ -296,12 +296,32 @@ class Agent {
     this.agentServicesProps.signifyClient = this.signifyClient;
     await this.connectSignifyClient();
 
+    await this.basicStorage.save({
+      id: MiscRecordId.PROCESS_RECOVERING,
+      content: { value: true },
+    });
+
     await SecureStorage.set(KeyStoreKeys.SIGNIFY_BRAN, bran);
     await this.saveAgentUrls({
       url: connectUrl,
       bootUrl: "",
     });
     this.markAgentStatus(true);
```

Comment:
Hmm, what happens here in the UI though if the sync takes several seconds? Will the UI be loaded but half filled?

I would say we need to mark it as online after, and just allow this "sync" to happen when we're "offline" but really online.

Reply:
@iFergal because we have a decorator `@OnlineOnly` before `syncKeriaContacts`, `syncKeriaIdentifiers`, `syncACDCs` so I think we cannot sync if the status has not been online

Reply:
Yeah, what I mean is we can remove that decorator. Perhaps then we can handle the Signify error in the UI differently